### PR TITLE
Remove explicit calls to Commit when using ReadWriteTransaction

### PR DESCRIPTION
**Pull request by AlCutter**
The `ReadWriteTransaction` method will always call `Commit()` if no error is returned by the passed-in function, so this PR removes the explicit calls to `Commit` from the passed-in function implementations.

### CODE REVIEW

File: storage/memory/admin_storage.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func (s *memoryAdminStorage) ReadWriteTransaction(ctx context.Context, f storage
 	}
 	defer tx.Close()
 	if err := f(ctx, tx); err != nil {
-		tx.Close()
+		return err
```
Comment (Martin2112) [inline storage/memory/admin_storage.go]: Why did I miss that before?

File: storage/tools/dump_tree/dumplib/dumplib.go [hunk 2]
```diff
@@ -256,23 +255,22 @@ func Main(args Options) string {
 	sequenceLeaves(ls, seq, tree.TreeId, args.TreeSize, args.BatchSize, args.LeafFormat)
 
 	// Read the latest STH back
-	tx, err := ls.BeginForTree(context.TODO(), tree.TreeId)
-	if err != nil {
-		glog.Fatalf("BeginForTree got: %v, want: no err", err)
-	}
+	var sth trillian.SignedLogRoot
+	err := ls.ReadWriteTransaction(context.TODO(), tree.TreeId, func(ctx context.Context, tx storage.LogTreeTX) error {
```
Comment (Martin2112) [inline storage/tools/dump_tree/dumplib/dumplib.go]: SnapshotForTree? Doesn't really matter as this is test code really.

Comment (AlCutter) [reply storage/tools/dump_tree/dumplib/dumplib.go]: Yeah, I just converted it as-was.

File: storage/memory/admin_storage.go (hunk 1 above)
Comment (AlCutter) [reply storage/memory/admin_storage.go]: easy to miss in the noise :)

**Pull request merged**